osi_ippay_payment/models/payment.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import requests
+import xmltodict
+import json
+
+from odoo import api, fields, models, _
+from odoo.addons.payment.models.payment_acquirer import ValidationError
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    provider = fields.Selection(selection_add=[('ippay', 'IPpay')])
+
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    @api.model
+    def ippay_create(self, values):
+
+        if values.get('cc_number'):
+            values['cc_number'] = values['cc_number'].replace(' ', '')
+            card_detail = {'cc_number': values['cc_number'],
+                           'expiry_month': values['cc_expiry_month'],
+                           'expiry_year': values['cc_expiry_year']}
+
+            xml = '''<ippay>
+            <TransactionType>TOKENIZE</TransactionType>
+            <TerminalID>TESTTERMINAL</TerminalID>
+            <CardNum>%s</CardNum>
+            <CardExpMonth>%s</CardExpMonth>
+            <CardExpYear>%s</CardExpYear>
+            </ippay>''' % (card_detail.get('cc_number'),
+                           card_detail.get('expiry_month'),
+                           card_detail.get('expiry_year'))
+
+            r = requests.post('https://testgtwy.ippay.com/ippay',
+                              data=xml, headers={'Content-Type': 'text/xml'})
+            data = eval(json.dumps(xmltodict.parse(
```

Comment:
No need for `eval(json.dumps(`:
the result from xmltodict.parse is already a Dict-like object (an OrderedDict, to be precise).